bundled/tool/lsp_server.py

```diff
@@ -142,6 +142,14 @@ def _get_group_dict(line: str) -> Optional[Dict[str, str]]:
     return None
 
 
+def _build_message_doc_url(code: str) -> str:
+    """Build the URL to the documentation for this warning."""
+    anchor = utils.ERROR_CODE_ANCHORS.get(code)
+    if anchor is None:
+        return ""
+    return utils.ERROR_CODE_BASE_URL + anchor
```

Comment:
```suggestion
def _get_code_description(code: str) -> Opinion[lsp.CodeDescription]:
    """Build the URL to the documentation for this warning."""
    anchor = utils.ERROR_CODE_ANCHORS.get(code)
    if anchor is None:
        return None
    return lsp.CodeDescription(href=utils.ERROR_CODE_BASE_URL + anchor)
```

Reply:
I think you meant `Optional` ?

```suggestion
def _get_code_description(code: str) -> Optional[lsp.CodeDescription]:
    """Build the URL to the documentation for this warning."""
    anchor = utils.ERROR_CODE_ANCHORS.get(code)
    if anchor is None:
        return None
    return lsp.CodeDescription(href=utils.ERROR_CODE_BASE_URL + anchor)
```